README.md

```diff
@@ -34,19 +34,17 @@ class ssh {
 }
 
 class ssh::client {
-  noop(false)
+  noop(undef)
 
   file { '/etc/ssh/ssh_config':
     ensure => file,
   }
 }
 ```
 
-In the above example, none of the resources in `Class['ssh']` will be enforced
-but the resources in Class['ssh::client'] *WILL* be enforced because the
-default noop value is reset to false in the child's scope. Without `noop(false)`
-in `Class['ssh::client']`, the parent scope's default noop value (as set with the `noop()`
-function) would be inherited.
+In the above example, none of the resources in `Class['ssh']` will be enforced. The resources in Class['ssh::client'] *WILL* be enforced, because the default noop value is reset in the child's scope. Without `noop(undef)` in `Class['ssh::client']`, the parent scope's default noop value (as set with the `noop()` function) would be inherited.
+
+Another option would be to call `noop(false)` instead of `noop(undef)` in Class['ssh::client']. The difference between these two options is how the resources will behave if Puppet runs with the `--noop` setting or CLI argument. When `noop(undef)` is used to reset the noop default in Class['ssh::client'], the CLI `--noop` flag will still work to set these resources to noop. If `noop(false)` is used, however, these resources will never noop—not even when `--noop` is used on the command line.
```

Comment:
Could you format the puppet code Class statements here too.

Reply:
Sure thing. Lemme see if I can use Github's nifty suggestion feature to just do that here...

```suggestion
Another option would be to call `noop(false)` instead of `noop(undef)` in `Class['ssh::client']`. The difference between these two options is how the resources will behave if Puppet runs with the `--noop` setting or CLI argument. When `noop(undef)` is used to reset the noop default in `Class['ssh::client']`, the CLI `--noop` flag will still work to set these resources to noop. If `noop(false)` is used, however, these resources will never noop—not even when `--noop` is used on the command line.
```